parser/src/error.rs

```diff
@@ -0,0 +1,59 @@
+//! Define internal parse error types
+//! The goal is to provide a matching and a safe error API, maksing errors from LALR
+extern crate lalrpop_util;
+use self::lalrpop_util::ParseError as InnerError;
+
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug)]
+pub enum ParseError {
+    // Parser encountered an unexpected end of input
+    EOF,
+    // Parser encountered an extra token
+    ExtraToken,
+    // Parser encountered an invalid token
+    InvalidToken,
```

Comment:
This is handy! Can you add location information as well? So then we know where the error occurred.